sdk/core/azure-core/azure/core/pipeline/policies/_base_async.py

```diff
@@ -47,12 +50,17 @@ class AsyncHTTPPolicy(abc.ABC, Generic[HTTPRequestType, AsyncHTTPResponseType]):
 
     def __init__(self) -> None:
         # next will be set once in the pipeline
-        from ..transport._base_async import AsyncHttpTransport
 
-        self.next = cast(Union[AsyncHTTPPolicy, AsyncHttpTransport], None)
+        self.next = cast(
```

Comment:
It's the None problem again

Reply:
You should be able to not assign a `None` value and instead declare the instance variable per PEP526 if you can guarantee that the value isn't observed before it is set.

Reply:
LEt me try

Reply:
Nope, not working. Trying to see if I find a workaround

Reply:
So I can remove the `__init__` entirely, and now it's happy and pass mypy.

Impact is that if you were creating a policy without a pipeline and calling `next`, instead of a `AttributeError: 'NoneType' object has no attribute 'next'` now you would get `AttributeError: 'MyPolicy' object has no attribute 'next'.

Thoughts?